src/freenet/io/xfer/BlockTransmitter.java

```diff
@@ -156,24 +160,21 @@ public void run() {
 						// the variable count is used to count which message is being processed now
 						// the HTL is taken into consideration when adding delay too
 						count++;
-						if (isHighHtl() && count >= (Node.PACKETS_IN_BLOCK - 1)) {
-							try {
-								wait((int) (Math.random() * MAX_ARTIFICIAL_FINAL_PACKETS_DELAY));
-							} catch (InterruptedException e) {
-								// intentionally left blank: can continue
-							}
+						if (isHighHtl() && count >= (Node.PACKETS_IN_BLOCK - 2)) {
```

Comment:
Note that `Node.PACKETS_IN_BLOCK - 1` has changed to `Node.PACKETS_IN_BLOCK - 2` as we now wait after sending rather than before.